code/modules/admin/verbs/playsound.dm

```diff
@@ -151,17 +162,22 @@
 	if(web_sound_url && !findtext(web_sound_url, GLOB.is_http_protocol))
 		tgui_alert(user, "The media provider returned a content URL that isn't using the HTTP or HTTPS protocol. This is a security risk and the sound will not be played.", "Security Risk", list("OK"))
 		to_chat(user, span_boldwarning("BLOCKED: Content URL not using HTTP(S) Protocol!"), confidential = TRUE)
-
 		return
 	if(web_sound_url || stop_web_sounds)
-		for(var/m in GLOB.player_list)
-			var/mob/M = m
-			var/client/C = M.client
-			if(C.prefs.read_preference(/datum/preference/toggle/sound_midi))
+		var/list/players = GLOB.player_list
+		if (range)
+			players = list()
+			for (var/mob/listening_mob in get_hearers_in_view(range, user))
+				if (listening_mob.client)
+					players += listening_mob
+		for(var/player as anything in players)
+			var/mob/player_mob = player
+			var/client/player_client = player_mob.client
```

Comment:
not how as anything works. it isnt a free time save. its a time save by skipping internal istype checks. var/player isnt typed. the reason to convert it to as anything is that you can delete player_mob and type player

Reply:
Oh I see, will push in a second (I am too lazy to look at docs)